protocol/driver.go

```diff
@@ -190,3 +228,108 @@ func (m *ParseResponse) Nodes() (nodes.Node, error) {
 	}
 	return ast, err
 }
+
+// Version implements DriverHostClient.
+func (c *client) Version(rctx context.Context) (driver.Version, error) {
+	sp, ctx := opentracing.StartSpanFromContext(rctx, "bblfsh.client.Version")
+	defer sp.Finish()
+
+	resp, err := c.h.ServerVersion(ctx, &VersionRequest{})
+	if err != nil {
+		return driver.Version{}, err
+	}
+	vers := (*driver.Version)(resp.Version)
+	return *vers, nil
+}
+
+// Languages implements DriverHostClient.
+func (c *client) Languages(rctx context.Context) ([]manifest.Manifest, error) {
+	sp, ctx := opentracing.StartSpanFromContext(rctx, "bblfsh.client.Languages")
+	defer sp.Finish()
+
+	resp, err := c.h.SupportedLanguages(ctx, &SupportedLanguagesRequest{})
+	if err != nil {
+		return nil, err
+	}
+	out := make([]manifest.Manifest, 0, len(resp.Languages))
+	for _, m := range resp.Languages {
+		dm := m.ToNative()
```

Comment:
This just occurred to me: Do we need this to be exported? If not, we could make it construct the value in-place. I wouldn't do that in an exported API, but here we only do it to get a copy for packing into the slice. This could then be:

```go
out := make([]manifest.Manifest, len(resp.Languages))
for i, m := range resp.Languages {
  m.toNative(&out[i])
}
```

assuming we had written

```go
func (m *Manifest) toNative(out *manifest.Manifest) {
   …
}
```

That would let us drop the extra copy and indirection.

Reply:
Good idea! I'd still leave the exported one as well (could be useful in the Go client), and make a new one unexported.

Reply:
Done!